File: code/413.cpp
class Solution {
public:
    int reverseInteger(int n) {
        long result = 0;
        do {
            int i = n % 10;
            result = result * 10 + i;
            n /= 10;
        } while (n != 0);
        if (result > INT_MAX)
            return 0;
        else
            return result;
    }
};

File: Python3/413.py
'''
Created on 2018年2月28日

@author: Li
'''
class Solution:
    """
    @param n: the integer to be reversed
    @return: the reversed integer
    """
    def reverseInteger(self, n):
        # write your code here
        fun = lambda x: int(str(x)[::-1])
        if n >= 0:
            n = fun(n)
        else:
            n = -1 * fun(-1 * n)
        result = n if n < 2147483647 else 0
        return result
            
